### ISSUE TITLE
CSS Class Name on <super-navbar-button>

### ISSUE THREAD

**Issue by mikeCatalyst** [AppGyver/steroids#756]
Setting a class (or id) on the webcomponent does not seem to flow through when the native component is created. Any style set in ios.css on that class is not seen when the app runs.

This works with <super-navbar> however.

**Comment by harsaharsa** [comment]
That's actually a not-implemented feature. 

I noticed `supersonic.ui.NavigationBarButton` was missing docs about the style options, so they're now here: http://docs.appgyver.com/supersonic/api-reference/stable/supersonic/ui/navigationbarbutton-class/

However, the `<super-navbar-button>` code doesn't monitor changes, see e.g. how `<super-navbar>` does it:

https://github.com/AppGyver/supersonic/blob/master/components/super-navbar/super-navbar.coffee#L143-L145

We'll put it on the backlog, and as always, PRs are welcome!

**Comment by mikeCatalyst** [comment]
Hm, I'm confused then. In your documentation it says you can define style classes and id's for all native element. (http://docs.appgyver.com/supersonic/guides/ui/styling-native-components/).

This is all I'm trying to do, adding a class to my super-navbar-button, and having a style affect just that single button. The style is declared in ios.css.

example:

<super-navbar-button class="myClass">text</super-navbar-button>

Here it looks like the navbar-button is looking for a defined class attribute .. https://github.com/AppGyver/supersonic/blob/master/components/super-navbar-button/super-navbar-button.coffee#L108

**Comment by harsaharsa** [comment]
Yes – it's a bug that it's not implemented (and a rather embarrassing one at that, which I apologize for). The code you quoted is half implemented – it listens for changes in all of those attributes, but as you can see from [this bit](https://github.com/AppGyver/supersonic/blob/master/components/super-navbar-button/super-navbar-button.coffee#L29-L46), nothing is done for the class/id/style attribute changes.  We'll work on a fix ASAP.

**Comment by mikeCatalyst** [comment]
Great news, thanks for the help!

**Comment by timmynz** [comment]
Has his been fixed?

**Comment by pre** [comment]
Not in Supersonic, but as a workaround, you can call `setClass()` directly. http://docs.appgyver.com/supersonic/api-reference/stable/supersonic/ui/navigationbar/setclass/

**Comment by budda** [comment]
> We'll work on a fix ASAP.

Any idea on roadmap to implementing this ?
